examples/fleet/main.go

```diff
@@ -185,7 +186,10 @@ func (k *KindClusterProvider) Run(ctx context.Context, mgr manager.Manager) erro
 				k.log.Info("failed to create rest config", "error", err)
 				return false, nil // keep going
 			}
-			cl, err := cluster.New(cfg, k.Options...)
+			// Copy provider options and append the cluster name
+			clOptions := append(k.Options, cluster.WithName(clusterName))
+
+			cl, err := cluster.New(cfg, clOptions...)
```

Comment:
this is technically not copying.
```suggestion
			clOptions := append([]cluster.Option{cluster.WithName(clusterName)}, k.Options...)
			cl, err := cluster.New(cfg, clOptions...)
```

Reply:
Compare https://goplay.tools/snippet/EkluEYuqP01

Reply:
Good catch! Thank you for fixing